crates/bindings/c/src/lib.rs

```diff
@@ -116,10 +126,124 @@ pub unsafe extern "C" fn libsql_column_count(res: libsql_rows_t) -> std::ffi::c_
 }
 
 #[no_mangle]
-pub unsafe extern "C" fn libsql_value_text(
-    _res: libsql_rows_t,
-    _row: std::ffi::c_int,
-    _col: std::ffi::c_int,
+pub unsafe extern "C" fn libsql_column_name(
+    res: libsql_rows_t,
+    col: std::ffi::c_int,
+) -> *const std::ffi::c_char {
+    let res = res.get_ref();
+    if col >= res.column_count() {
+        return std::ptr::null();
+    }
+    let name = res.column_name(col);
+    match std::ffi::CString::new(name) {
+        Ok(name) => name.into_raw(),
+        Err(_) => std::ptr::null(),
+    }
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn libsql_column_type(
+    res: libsql_rows_t,
+    col: std::ffi::c_int,
+) -> std::ffi::c_int {
+    let res = res.get_ref();
+    if col >= res.column_count() {
+        return -1;
+    }
+    res.column_type(col)
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn libsql_next_row(res: libsql_rows_t) -> libsql_row_t {
+    if res.is_null() {
+        return libsql_row_t::null();
+    }
+    let res = res.get_ref();
+    match res.next() {
+        Ok(Some(row)) => {
+            let row = Box::leak(Box::new(libsql_row { result: row }));
+            libsql_row_t::from(row)
```

Comment:
Cool. I was following existing code in this file. Will change all the places that use Box::leak. Thanks.

Reply:
It does not seem to work.
if I do:
```
-            let row = Box::leak(Box::new(libsql_row { result: row }));
-            libsql_row_t::from(row)
+            // let row = Box::leak(Box::new(libsql_row { result: row }));
+            // libsql_row_t::from(row)
+            Box::<libsql_row>::into_raw(Box::new(libsql_row { result: row }))

```
I get:
```
error[E0308]: mismatched types
   --> bindings/c/src/lib.rs:166:13
    |
157 | ...e extern "C" fn libsql_next_row(res: libsql_rows_t) -> libsql_row_...
    |                                                           ------------ expected `libsql_row_t` because of return type
...
166 | ...   Box::<libsql_row>::into_raw(Box::new(libsql_row { result: row }))
    |       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `libsql_row_t`, found `*mut libsql_row`
    |
    = note:   expected struct `libsql_row_t`
            found raw pointer `*mut libsql_row`
```

Reply:
ah, probably because `libsql_row_t` is defined as `*const libsql_row`. Weird that mut doesn't get coerced to const automatically, but I'm pretty sure if it would work if you make it a `Box::into_raw(...) as *const _`